File: MiniDOMWebCtrls/DMD/CQSPD/MotivazioniPraticheAnnullateModuleHandler.cs
using System;
using DMD.XML;

namespace minidom.Forms
{
    public class MPANNHSTATSITEM 
        : IComparable, IDMDXMLSerializable
    {
        public int Count = 0;
        public decimal ML = 0m;
        public string Motivo = "";

        public MPANNHSTATSITEM()
        {
            DMDObject.IncreaseCounter(this);
        }

        public int CompareTo(object obj)
        {
            MPANNHSTATSITEM other = (MPANNHSTATSITEM)obj;
            if (ML > other.ML)
            {
                return 1;
            }
            else if (ML < other.ML)
            {
                return -1;
            }
            else
            {
                return 0;
            }
        }

        protected virtual void SetFieldInternal(string fieldName, object fieldValue)
        {
            switch (fieldName ?? "")
            {
                case "Count":
                    {
                        Count = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                        break;
                    }

                case "ML":
                    {
                        ML = (decimal)DMD.XML.Utils.Serializer.DeserializeDouble(DMD.Strings.CStr(fieldValue));
                        break;
                    }

                case "Motivo":
                    {
                        Motivo = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                        break;
                    }
            }
        }

        protected virtual void XMLSerialize(XMLWriter writer)
        {
            writer.WriteAttribute("Count", Count);
            writer.WriteAttribute("ML", ML);
            writer.WriteAttribute("Motivo", Motivo);
        }

        void IDMDXMLSerializable.XMLSerialize(XMLWriter writer)
        {
            this.XMLSerialize(writer);
        }

        void IDMDXMLSerializable.SetFieldInternal(string fieldName, object fieldValue)
        {
            this.SetFieldInternal(fieldName, fieldValue);
        }


        ~MPANNHSTATSITEM()
        {
            DMDObject.DecreaseCounter(this);
        }
    }

    public class MotivazioniPraticheAnnullateModuleHandler : CQSPDBaseStatsHandler
    {
        public MotivazioniPraticheAnnullateModuleHandler()
        {
        }

        public override Databases.DBObjectCursorBase CreateCursor()
        {
            return new Finanziaria.CPraticheCQSPDCursor();
        }
    }
}

File: MiniDOMWebCtrls/DMD/CQSPD/MotivazioniPraticheAnnullateModuleHandler.vb
Imports minidom
Imports minidom.Forms
Imports minidom.Databases
Imports minidom.WebSite

Imports minidom.Finanziaria
Imports minidom.Sistema
Imports minidom.Anagrafica

Namespace Forms

    Public Class MPANNHSTATSITEM
        Implements IComparable, XML.IDMDXMLSerializable


        Public Count As Integer = 0
        Public ML As Decimal = 0
        Public Motivo As String = ""

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
        End Sub

        Public Function CompareTo(obj As Object) As Integer Implements IComparable.CompareTo
            Dim other As MPANNHSTATSITEM = obj
            If (Me.ML > other.ML) Then
                Return 1
            ElseIf (Me.ML < other.ML) Then
                Return -1
            Else
                Return 0
            End If
        End Function

        Protected Overridable Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "Count" : Me.Count = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "ML" : Me.ML = XML.Utils.Serializer.DeserializeDouble(fieldValue)
                Case "Motivo" : Me.Motivo = XML.Utils.Serializer.DeserializeString(fieldValue)
            End Select
        End Sub

        Protected Overridable Sub XMLSerialize(writer As XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("Count", Me.Count)
            writer.WriteAttribute("ML", Me.ML)
            writer.WriteAttribute("Motivo", Me.Motivo)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class

    Public Class MotivazioniPraticheAnnullateModuleHandler
        Inherits CQSPDBaseStatsHandler

        Public Sub New()

        End Sub



        Public Overrides Function CreateCursor() As DBObjectCursorBase
            Return New CPraticheCQSPDCursor
        End Function

    End Class


End Namespace